aws/resource_aws_eks_cluster.go

```diff
@@ -236,7 +245,32 @@ func resourceAwsEksClusterUpdate(d *schema.ResourceData, meta interface{}) error
 
 		err = waitForUpdateEksCluster(conn, d.Id(), updateID, d.Timeout(schema.TimeoutUpdate))
 		if err != nil {
-			return fmt.Errorf("error waiting for EKS Cluster (%s) update (%s): %s", d.Id(), updateID, err)
+			return fmt.Errorf("error waiting for EKS Cluster (%s) version update (%s): %s", d.Id(), updateID, err)
+		}
+	}
+
+	if d.HasChange("vpc_config") {
+		input := &eks.UpdateClusterConfigInput{
+			Name:               aws.String(d.Id()),
+			ResourcesVpcConfig: expandEksVpcConfigUpdateRequest(d.Get("vpc_config").([]interface{})),
+		}
+
+		log.Printf("[DEBUG] Updating EKS Cluster (%s) config: %s", d.Id(), input)
+		output, err := conn.UpdateClusterConfig(input)
+
+		if err != nil {
+			return fmt.Errorf("error updating EKS Cluster (%s) config: %s", d.Id(), err)
+		}
+
+		if output == nil || output.Update == nil || output.Update.Id == nil {
+			return fmt.Errorf("error determining EKS Cluster (%s) config update ID: empty response", d.Id())
+		}
+
+		updateID := aws.StringValue(output.Update.Id)
+
+		err = waitForUpdateEksCluster(conn, d.Id(), updateID, d.Timeout(schema.TimeoutUpdate))
```

Comment:
We should probably add a note for the `update` customizable timeout that it is being used separately for both version and configuration update timeouts.

Reply:
Yes,  I was thinking of that. Since both version and network access updates could happen in the same `terraform apply` do we want to up the default `Update` timeout?
A change to just the network access took ~4mins in my testing.

Reply:
Each usage of the timeout duration is separate, as far as I know, so we shouldn't need to increase it.

Reply:
Yes, realized that after I wrote my comment.